homeassistant/components/hassio.py

```diff
@@ -0,0 +1,262 @@
+        if answer['result'] == 'ok':
+            return answer['data'] if answer['data'] else True
+
+        _LOGGER.error("%s return error %s.", cmd, answer['message'])
+        return False
+
+    @asyncio.coroutine
+    def send_raw(self, cmd, payload=None):
+        """Send raw request to API."""
+        try:
+            with async_timeout.timeout(TIMEOUT, loop=self.loop):
+                request = yield from self.websession.get(
+                    "http://{}{}".format(self._ip, cmd),
+                    timeout=None, json=payload
+                )
+
+                if request.status != 200:
+                    _LOGGER.error("%s return code %d.", cmd, request.status)
+                    return
+
+                return (yield from request.json())
+
+        except asyncio.TimeoutError:
+            _LOGGER.error("Timeout on api request %s.", cmd)
+
+        except aiohttp.ClientError:
+            _LOGGER.error("Client error on api request %s.", cmd)
+
+
+class HassIOBaseView(HomeAssistantView):
+    """HassIO view to handle base part."""
+
+    requires_auth = True
+
+    def __init__(self, hassio, base):
+        """Initialize a hassio base view."""
+        self.hassio = hassio
+        self._url_info = "/{}/info".format(base)
+
+        self.url = "/api/hassio/{}".format(base)
+        self.name = "api:hassio:{}".format(base)
+
+    @asyncio.coroutine
+    def get(self, request):
+        """Get base data."""
+        data = yield from self.hassio.send_command(self._url_info)
+        if not data:
+            raise HTTPBadGateway()
+        return web.json_response(data)
+
+
+class HassIOBaseEditView(HassIOBaseView):
+    """HassIO view to handle base with options support."""
+
+    def __init__(self, hassio, base):
+        """Initialize a hassio base edit view."""
+        super().__init__(hassio, base)
+        self._url_options = "/{}/options".format(base)
+
+    @asyncio.coroutine
+    def post(self, request):
+        """Set options on host."""
+        data = yield from request.json()
+
+        response = yield from self.hassio.send_raw(
+            self._url_options, payload=data)
+        if not response:
+            raise HTTPBadGateway()
+        return web.json_response(response)
+
+
+class HassIOAddonsView(HomeAssistantView):
+    """HassIO view to handle addons part."""
+
+    requires_auth = True
+    url = "/api/hassio/addons/{addon}"
+    name = "api:hassio:addons"
+
+    def __init__(self, hassio):
+        """Initialize a hassio addon view."""
+        self.hassio = hassio
+
+    @asyncio.coroutine
+    def get(self, request, addon):
+        """Get addon data."""
+        data = yield from self.hassio.send_command(
+            "/addons/{}/info".format(addon))
+        if not data:
+            raise HTTPBadGateway()
+        return web.json_response(data)
+
+    @asyncio.coroutine
+    def post(self, request, addon):
+        """Set options on host."""
+        data = yield from request.json()
```

Comment:
I think we make only a passthrought. HassIO will give a error back to frontend if a option will not support it.